docs/config.md
---
navigation: 5
---

# Configuration

#### Show Information


| property        | description       |
|-----------------|-------------------|
| show.title      | Show Title        |
| show.subtitle   | Show Description  |
| show.summary    | Show Summary      |
| show.poster     | Show Cover        |
| show.link       | Url to show       |

```javascript
{
    show: {
      title: 'Freak Show',
      subtitle: 'Menschen! Technik! Sensationen!',
      summary: 'Die muntere Talk Show um Leben mit Technik, das Netz und Technikkultur. Bisweilen Apple-lastig aber selten einseitig. Wir leben und lieben Technologie und reden darüber. Mit Tim, hukl, roddi, Clemens und Denis. Freak Show hieß irgendwann mal mobileMacs.',
      poster: 'https://freakshow.fm/wp-content/cache/podlove/04/662a9d4edcf77ea2abe3c74681f509/freak-show_200x200.jpg',
      link: 'https://freakshow.fm'
    }
}
```

#### Episode Information

| property        | description               |
|-----------------|---------------------------|
| title           | Episode Title             |
| subtitle        | Episode Description       |
| summary         | Episode Summary           |
| poster          | Episode Cover             |
| publicationDate | Episode Publication Date  |
| duration        | Episode Duration          |
| link            | Link to Episode           |

```javascript
{
  title: 'FS171 Invasion!',
  subtitle: 'LAN Planung - Kalender - Bingo - Wikipedia - Akkukalibration - Alte iPads und iPods - Find My Friends - iPhone Music Player - Apple Watch - Kommandozeile - Star Wars - Dante - Internet of Things Security - VPN',
  summary: 'Wir haben eine wie wir finden abwechslungsreiche Sendung produziert, die wir Euch wie immer mit Freude bereitstellen. Während die Live-Hörer Freak-Show-Bingo spielen, greifen wir das Wikipedia-Thema der letzten Sendung auf und liefern auch noch weitere Aspekte des optimalen Star-Wars-Medienkonsums frei Haus. Dazu viel Nerderei rund um die Kommandozeile, eine Einschätzung der Perspektive der Apple Watch, ein Rant über die mangelhafte Security  im Internet of Things (and Buildings) und allerlei anderer Kram.  Roddi setzt dieses Mal aus, sonst Vollbesetzung.',
  publicationDate: '2016-02-11T03:13:55+00:00',
  poster: 'https://freakshow.fm/wp-content/cache/podlove/04/662a9d4edcf77ea2abe3c74681f509/freak-show_200x200.jpg',
  duration: '04:15:32', // Simple time format
  link: 'https://freakshow.fm/fs171-invasion'
}
```

#### Episode Chapters

| property  | description                           |
|-----------|---------------------------------------|
| start     | Chapter Start Time in Format hh:mm:ss |
| title     | Chapter Title                         |
| href      | Chapter Link URL (optional)           |
| image     | Chapter Image URL (optional)          |

_Not providing chapters will disable all chapter related functions._

```javascript
{
    chapters: [
      { start:"00:00:00", title: 'Intro'},
      { start:"00:01:39", title: 'Begrüßung'},
      { start:"00:04:58", title: 'IETF Meeting Netzwerk'},
      { start:"00:18:37", title: 'Kalender'},
      { start:"00:33:40", title: 'Freak Show Bingo', image: 'https://picsum.photos/400/400/?random&image=6'},
      { start:"00:35:37", title: 'Wikipedia', href: 'https://wikipedia.de'},
      { start:"01:17:26", title: 'iPhone Akkukalibration'},
      { start:"01:24:55", title: 'Alte iPads und iPod touches'},
      { start:"01:31:02", title: 'Find My Friends'},
      { start:"01:41:46", title: 'iPhone Music Player'},
      { start:"01:56:13", title: 'Apple Watch'},
      { start:"02:11:51", title: 'Kommandozeile: System Appreciation'},
      { start:"02:23:10", title: 'Sound und Design für Games'},
      { start:"02:24:59", title: 'Kommandozeile: Remote Deployment'},
      { start:"02:32:37", title: 'Kommandozeile: Man Pages'},
      { start:"02:44:31", title: 'Kommandozeile: screen vs. tmux'},
      { start:"02:58:02", title: 'Star Wars: Machete Order & Phantom Edit'},
      { start:"03:20:05", title: 'Kopfhörer-Ersatzteile'},
      { start:"03:23:39", title: 'Dante'},
      { start:"03:38:03", title: 'Dante Via'},
      { start:"03:45:33", title: 'Internet of Things Security'},
      { start:"03:56:11", title: 'That One Privacy Guy\'s VPN Comparison Chart'},
      { start:"04:10:00", title: 'Ausklang'}
    ]
}
```

#### Audio Files

```javascript
{
  audio: [{
    url: 'http://freakshow.fm/podlove/file/4468/s/download/c/select-show/fs171-invasion.m4a',
    mimeType: 'audio/mp4',
    size: 93260000,
    title: 'Audio MP4'
  }, {
    url: 'http://freakshow.fm/podlove/file/4467/s/download/c/select-show/fs171-invasion.mp3',
    mimeType: 'audio/mpeg',
    size: 14665000,
    title: 'Audio MP3'
  }, {
    url: 'http://freakshow.fm/podlove/file/4467/s/download/c/select-show/fs171-invasion.oga',
    mimeType: 'audio/ogg',
    size: 94400000,
    title: 'Audio Ogg'
  }, {
    url: 'http://freakshow.fm/podlove/file/4467/s/download/c/select-show/fs171-invasion.opus',
    mimeType: 'audio/opus',
    size: 94400000,
    title: 'Audio Opus'
  }]
}
```

#### References

| property         | description                                                                   |
|------------------|-------------------------------------------------------------------------------|
| reference.base   | Reference to webplayer base, if not provided it falls back to the current url |
| reference.config | Reference to the current configuration                                        |
| reference.share  | Reference sharing embed endpoint                                              |

```javascript
{
    reference: {
      base: '//podlove-player.surge.sh',
      config: '//podlove-player.surge.sh/fixtures/example.json',
      share: '//podlove-player.surge.sh/share'
    }
}
```

#### Theming

| property        | description                         |
|-----------------|-------------------------------------|
| theme.main      | main theme color (default: #2B8AC6) |
| theme.highlight | highlight theme color               |

```javascript
{
    theme: {
      main: '#2B8AC6',
      highlight: '#EC79F2'
    }
}
```

#### Runtime

| property         | description                                          |
|------------------|------------------------------------------------------|
| runtime.platform | Platform (desktop,mobile), is detected by the player |
| runtime.language | Language (en,de,...), defaults to browser language   |

```javascript
{
    runtime: {
      platform: 'desktop',
      language: 'en'
    }
}
```

#### Contributors

| property           | description                          |
|--------------------|--------------------------------------|
| contributor.name   | Name of contributor (e.g. speaker)   |
| contributor.avatar | Absolute Url to contributor's avatar |

```javascript
 contributors: [{
      id: 'tim-pritlove',
      avatar: 'https://freakshow.fm/wp-content/cache/podlove/47/08928e3c26dcb1141d67ad75869619/tim-pritlove_50x50.jpg',
      name: 'Tim Pritlove'
  }, {
      id: 'clemens-schrimpe',
      avatar: 'https://freakshow.fm/wp-content/cache/podlove/0f/9c18f5e825496b9060337f92814142/clemens-schrimpe_50x50.jpg',
      name: 'Clemens Schrimpe'
  }, {
      id: 'hukl',
      avatar: 'https://freakshow.fm/wp-content/cache/podlove/8e/f30cbe274c3f5e43dc4a7219676f50/hukl_50x50.jpg',
      name: 'hukl'
  }, {
      id: 'denis-ahrens',
      avatar: 'https://freakshow.fm/wp-content/cache/podlove/b2/425e5c8f180ddf548c95be1c2d7bcf/denis-ahrens_50x50.jpg',
      name: 'Denis Ahrens'
  }]
```

#### Transcripts

To enable transcripts provide a URI to the transcriptions file.

| property    | description                      |
|-------------|----------------------------------|
| transcripts | `http://url.to/transcripts.json` |

The transcriptions file contains transcripts in the following format:

```javascript
[{
  "start": "00:00:43.1", // Simple Time Format
  "end": "00:01:17.2", // Simple Time Format
  "speaker?": "Speaker ID",
  "test": "Transcribed text"
}]
```

#### Tabs

| property      | description             |
|---------------|-------------------------|
| tabs.info     | toggle the info tab     |
| tabs.share    | toggle the share tab    |
| tabs.chapters | toggle the chapters tab |
| tabs.audio    | toggle the audio tab    |
| tabs.download | toggle the download tab |

```javascript
  tabs: {
   info: true // will expand the info tab on load
  }
```

#### Visible Components

List of components that are visible in the player. if nothing is provided all components are visible.

| property          | description           |
|-------------------|-----------------------|
| showTitle         | Header show title     |
| episodeTitle      | Header episode title  |
| subtitle          | Header subtitle       |
| progressbar       | Play progress         |
| controlSteppers   | Steppers controls     |
| controlChapters   | Chapters controls     |
| poster            | Poster in Head        |
| tabChapters       | Chapters Tab          |
| tabFiles          | Files Tab             |
| tabAudio          | Audio Tab             |
| tabShare          | Share Tab             |


```javascript
  visibleComponents: [
    'tabInfo',
    'tabChapters',
    'tabFiles',
    'tabAudio',
    'tabShare',
    'poster',
    'showTitle',
    'episodeTitle',
    'subtitle',
    'progressbar',
    'controlSteppers',
    'controlChapters'
  ]
```

#### Simple Time Definitions

| example       | description                             |
|---------------|-----------------------------------------|
| `01:35:52`    | 1 hour 35 minutes 52 seconds            |
| `7:48`        | 7 minutes, 48 seconds                   |
| `35:12.250`   | 35 minutes, 12 seconds, 250 ms          |
| `05:12:03.5`  | 5 hours, 12 minutes, 3 seconds, 500 ms  |
| `37`          | 37 seconds                              |


CHANGELOG.md
## [4.2.4](https://github.com/podlove/podlove-web-player/compare/v4.2.3...v4.2.4) (2018-12-01)


### Bug Fixes

* **runtime:** merges initial configuration with runtime ([f7d689f](https://github.com/podlove/podlove-web-player/commit/f7d689f))
* **tabs:** changed the audio icon to the unified tab icon style ([bf2c64b](https://github.com/podlove/podlove-web-player/commit/bf2c64b))
* **transcripts:** better transcript end detection ([cd4f7f9](https://github.com/podlove/podlove-web-player/commit/cd4f7f9))


### Features

* **tabs:** change audio-tab icon to speaker symbol ([cfc427c](https://github.com/podlove/podlove-web-player/commit/cfc427c))
* **tabs:** dynamic audio icon based on volume ([107ffa7](https://github.com/podlove/podlove-web-player/commit/107ffa7))



## [4.2.3](https://github.com/podlove/podlove-web-player/compare/v4.2.2...v4.2.3) (2018-11-10)


### Bug Fixes

* **marquee:** marquee effect now doesn't produce unwanted spacing ([2279476](https://github.com/podlove/podlove-web-player/commit/2279476))



## [4.2.2](https://github.com/podlove/podlove-web-player/compare/v4.2.1...v4.2.2) (2018-11-10)



<a name="4.2.1"></a>
## [4.2.1](https://github.com/podlove/podlove-web-player/compare/v4.2.0...v4.2.1) (2018-11-09)


### Bug Fixes

* **safari:** uses fallback to native audio element ([ba5d2e9](https://github.com/podlove/podlove-web-player/commit/ba5d2e9))
* **styles:** resets number input in firefox and sets width of input-state ([e797522](https://github.com/podlove/podlove-web-player/commit/e797522))



# [4.2.0](https://github.com/podlove/podlove-web-player/compare/v4.1.7...v4.2.0) (2018-11-08)


### Bug Fixes

* **package:** update color to version 3.1.0 ([1d2d304](https://github.com/podlove/podlove-web-player/commit/1d2d304))
* **package:** update iframe-resizer to version 3.6.3 ([d838346](https://github.com/podlove/podlove-web-player/commit/d838346))
* **package:** update query-string to version 6.2.0 ([454ce84](https://github.com/podlove/podlove-web-player/commit/454ce84))
* **package:** update redux-actions to version 2.6.3 ([09257b9](https://github.com/podlove/podlove-web-player/commit/09257b9))
* **package:** update redux-actions to version 2.6.4 ([8ea43de](https://github.com/podlove/podlove-web-player/commit/8ea43de))
* **package:** update vue-i18n to version 8.2.1 ([d5552c0](https://github.com/podlove/podlove-web-player/commit/d5552c0)), closes [#795](https://github.com/podlove/podlove-web-player/issues/795)
* **package:** update vue-i18n to version 8.3.0 ([0710364](https://github.com/podlove/podlove-web-player/commit/0710364))
* **transcripts:** fixes searchbar button alignment in transcript header ([aac2fab](https://github.com/podlove/podlove-web-player/commit/aac2fab))


### Features

* **cleanup-embed:** Cleans embed on init and restores it in case of an error ([43e0c09](https://github.com/podlove/podlove-web-player/commit/43e0c09))
* **files:** adds files tab ([95d92e7](https://github.com/podlove/podlove-web-player/commit/95d92e7))
* **tabs/share, icon:** Add a linkedin share button ([f77fb4b](https://github.com/podlove/podlove-web-player/commit/f77fb4b))



<a name="4.1.7"></a>
## [4.1.7](https://github.com/podlove/podlove-web-player/compare/v4.1.6...v4.1.7) (2018-10-09)


### Bug Fixes

* **bundle:** Use bundled version of audio-driver ([0d022c9](https://github.com/podlove/podlove-web-player/commit/0d022c9))



<a name="4.1.6"></a>
## [4.1.6](https://github.com/podlove/podlove-web-player/compare/v4.1.5...v4.1.6) (2018-09-30)


### Bug Fixes

* add FiraSans-Bold (fira sans v8 600) webfont to improve rendering of bold in safari, change scss for chapter links to explicitly use bold when active ([fe856a8](https://github.com/podlove/podlove-web-player/commit/fe856a8))
* **chapters:** fix alt click interaction over the link part by making it based on the progressContainer vs. the event target ([f52e723](https://github.com/podlove/podlove-web-player/commit/f52e723))
* **chapters:** remove wrong click guard on onChapterClick, change to use linkHover state to guard to be in line with what is displayed ([8e3a3e0](https://github.com/podlove/podlove-web-player/commit/8e3a3e0))
* **package:** update detect-browser to version 3.0.1 ([e035721](https://github.com/podlove/podlove-web-player/commit/e035721))
* **package:** update iframe-resizer to version 3.6.2 ([2fb175e](https://github.com/podlove/podlove-web-player/commit/2fb175e))
* **package:** update lodash to version 4.17.11 ([df4cded](https://github.com/podlove/podlove-web-player/commit/df4cded))
* **package:** update lunr to version 2.3.3 ([ba4042e](https://github.com/podlove/podlove-web-player/commit/ba4042e))
* **package:** update mobile-detect to version 1.4.3 ([d1e74ba](https://github.com/podlove/podlove-web-player/commit/d1e74ba))
* **package:** update vue-i18n to version 8.1.0 ([4a13ecb](https://github.com/podlove/podlove-web-player/commit/4a13ecb))
* **tabs:** remove false 0.5 opacity of transcript icon, unify tab icon stroke widths and positions ([acc4fde](https://github.com/podlove/podlove-web-player/commit/acc4fde))


### Features

* **chapters:** add support for the "href" property in chapters by showing it in the chapters tab ([f7240f2](https://github.com/podlove/podlove-web-player/commit/f7240f2))



<a name="4.1.5"></a>
## [4.1.5](https://github.com/podlove/podlove-web-player/compare/v4.1.4...v4.1.5) (2018-08-25)


### Bug Fixes

* **docs:** Updates base for scripts ([d8b08dd](https://github.com/podlove/podlove-web-player/commit/d8b08dd))



<a name="4.1.4"></a>
## [4.1.4](https://github.com/podlove/podlove-web-player/compare/v4.1.3...v4.1.4) (2018-08-25)


### Bug Fixes

* **documentation:** Refactor docs ([423d811](https://github.com/podlove/podlove-web-player/commit/423d811))
* **package:** update detect-browser to version 3.0.0 ([7ce43dd](https://github.com/podlove/podlove-web-player/commit/7ce43dd))

podlove-web-player/settings.php
<?php

function podlovewebplayer_settings_init() {
	if ( is_admin() ){
		global $blog_id;
		$wp_options = get_option('podlovewebplayer_options');
		wp_enqueue_style( 'podlovewebplayer', plugins_url('static/podlove-web-player.css', __FILE__), array(), '2.1.0' );
		// wp_enqueue_style( 'pwpdesigner', plugins_url('static/libs/pwpdesigner/style.css', __FILE__), array(), '2.1.0' );
		// if(isset($wp_options['style_custom'])) {
		//   if($wp_options['style_custom'] !== '') {
		//     wp_enqueue_style( 'custom-pwp-style', plugins_url('static/customcss/pwp_custom_id-'.$blog_id.'.css', __FILE__), array(), $wp_options['style_version'] );
		//   }
		// } else {
		//   wp_dequeue_style( 'custom-pwp-style');
		// }
		// wp_enqueue_script( 'colorconverter', plugins_url('static/libs/pwpdesigner/colorconv.js', __FILE__), array(), '2.1.0' );
		// wp_enqueue_script( 'pwpdesigner', plugins_url('static/libs/pwpdesigner/script.js', __FILE__), array(), '2.1.0' );
		add_action( 'admin_menu', 'podlovewebplayer_create_menu' );
		add_action( 'admin_init', 'podlovewebplayer_register_settings' );
	}
}
add_action('init', 'podlovewebplayer_settings_init');

function podlovewebplayer_settings_page() { ?>
	<div class="wrap">
		<h2>Podlove Web Player Options</h2>
		<form method="post" action="options.php">
			<?php settings_fields('podlovewebplayer_options'); ?>
			<?php do_settings_sections('podlovewebplayer'); ?>
			<p class="submit">
			 <input name="Submit" type="submit" class="button button-primary" value="<?php esc_attr_e('Save Changes'); ?>" />
			</p>
		</form>
	</div>
<?php
}


function podlovewebplayer_css_path() {
	global $blog_id;
	$cssid = '_id-'.$blog_id;
	return plugin_dir_path(__FILE__) . "static/customcss/pwp_custom" . $cssid . ".css";
}

function podlovewebplayer_css_url() {
	global $blog_id;
	$cssid = '_id-'.$blog_id;
	return plugin_dir_url(__FILE__) . "static/customcss/pwp_custom" . $cssid . ".css";
}

function podlovewebplayer_custompwpstyle() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['style_custom'])) {
		$options['style_custom'] = file_get_contents(plugin_dir_url(__FILE__) . 'static/default.css');
	}
	$customstyle = $options['style_custom'];
	return $customstyle;
}

function podlovewebplayer_makecss() {
	if(chmod(plugin_dir_path(__FILE__) . "static/customcss/",0755)) {
		$podlovewebplayer_makecss = file_put_contents(podlovewebplayer_css_path(), "/* PodloveWebPlayer Custom Style */\n\n" . podlovewebplayer_custompwpstyle());
		return $podlovewebplayer_makecss;
	}
}


function podlovewebplayer_create_menu() {
	add_options_page(
		'Podlove Web Player Options',
		'Podlove Web Player',
		'manage_options',
		'podlovewebplayer',
		'podlovewebplayer_settings_page'
	);
}

function podlovewebplayer_register_settings() {
	$pwp = 'podlovewebplayer';
	$settings = array(
		'main' => array(
			'title' => 'General defaults',
			'fields' => array(
				'poster' => 'Poster URL'
			)
		),
		'audio' => array(
			'title'  => 'Audio player defaults',
			'fields' => array(
				'width'  => 'Audio width',
				'height' => 'Audio height',
				'type'   => 'Default MIME type'
			)
		),
		'video' => array(
			'title'  => 'Video player defaults',
			'fields' => array(
				'width'  => 'Video width',
				'height' => 'Video height',
				'type'   => 'Default MIME type'
			)
		),
		'enclosure' => array(
			'title'    => 'WordPress “enclosures”',
			'function' => true,
			'fields'   => array(
				'detect'     => 'Turn enclosures to players:',
				'force'      => 'Force enclosure:',
				'richplayer' => 'Advanced player:',
				'bottom'     => 'Put player to bottom of post:'
			)
		),
		'chapter' => array(
			'title' => 'Chapter',
			'fields' => array(
				'height' => 'Maximal Chapter height'
			),
		),
		// 'style' => array(
		// 'title'    => 'Player Style',
		// 'fields' => array(
		//   'custom'  => 'Style your Player:',
		//   'values'  => 'Designer Console:',
		//   'version' => 'Custon Style Version:'
		//   ),
		// ),
		'buttons' => array(
			'title'    => 'Buttons',
			'function' => true,
			'fields'   => array(
				'time'      => 'Hide time buttons:',
				'share'     => 'Hide share buttons:',
			)
		),
		'info' => array(
			'title'    => 'Information',
			'function' => true
		)
	);

	register_setting('podlovewebplayer_options', 'podlovewebplayer_options');

	foreach($settings as $sectionname => $section) {
		$function = false;
		if ( isset( $section['function'] ) ) {
			$function = $pwp.'_'.$sectionname;
		}
		add_settings_section( $pwp.'_'.$sectionname, $section['title'], $function, $pwp );
		if ( isset( $section['fields'] ) ) {
			$i = 0;
			foreach( $section['fields'] as $fieldname => $description ) {
				$i += 1;
				add_settings_field(
					$pwp.'_'.$sectionname.'_'.$fieldname,
					$description, $pwp.'_'.$sectionname.'_'.$fieldname,
					$pwp, $pwp.'_'.$sectionname,
					array( 'label_for' => 'pwp'.$sectionname.$i )
				);
			}
		}
	}
}

function podlovewebplayer_main_poster() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['main_poster'])) {
		$options['main_poster'] = "";
	}
	print '<textarea id="pwpposter" name="podlovewebplayer_options[main_poster]" rows="4" cols="50">' . $options['main_poster'] . '</textarea> <br><p>You can define default posters for your categories.<br>Write your defaults like this: <br><br>CategorieName=http://example.tld/url/of/image.png<br>NextCategorie=http://example.tld/anotherImg.png</p>';
}

function podlovewebplayer_audio_width() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['audio_width'])) {
		$options['audio_width'] = "";
	}
	print '<input id="pwpaudio1" name="podlovewebplayer_options[audio_width]" value="' . $options['audio_width'] . '" style="width:3em;" /> px&nbsp;&nbsp;(keep blank for automatic width)';
}

function podlovewebplayer_audio_height() {
	$options = get_option('podlovewebplayer_options');
	if (!isset( $options['audio_width'])) {
		$options['audio_height'] = '30';
	}
	print '<input id="pwpaudio2" name="podlovewebplayer_options[audio_height]" value="' . $options['audio_height'] . '" style="width:3em;" /> px&nbsp;&nbsp;(keep 30, if unsure)';
}

function podlovewebplayer_audio_type() {
	$options = get_option('podlovewebplayer_options');
	if ((!isset($options['audio_type'])) || ($options['audio_type'] == "audio/mp3")) {
		$options['audio_type'] = "audio/mpeg";
	}
	print '<input id="pwpaudio3" name="podlovewebplayer_options[audio_type]" value="' . $options['audio_type'] . '" style="width:6em;" />&nbsp;&nbsp;(such as \'audio/mpeg\')';
}

function podlovewebplayer_video_width() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['video_width'])) {
		$options['video_width'] = '640';
	}
	print '<input id="pwpvideo1" name="podlovewebplayer_options[video_width]" value="' . $options['video_width'] . '" style="width:3em" /> px';
}

function podlovewebplayer_video_height() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['video_height'])) {
		$options['video_height'] = '270';
	}
	print '<input id="pwpvideo2" name="podlovewebplayer_options[video_height]" value="' . $options['video_height'] . '" style="width:3em" /> px';
}

function podlovewebplayer_video_type() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['video_type'])) {
		$options['video_type'] = 'video/mp4';
	}
	print '<input id="pwpvideo3" name="podlovewebplayer_options[video_type]" value="' . $options['video_type'] . '" style="width:6em;" /> &nbsp;&nbsp;(such as \'video/mp4\')';
}

function podlovewebplayer_enclosure() {
	print "<p>WordPress automatically creates an \"enclosure\" custom field whenever it detects an URL to a media file in the post text.
		Use this option to turn these enclosures into Podlove Web Player instances.</p>\n\n
	<script>
	jQuery(function($){
		$('#pwpenclosure1').change(function(){
			if (this.checked == false) {
				$('#pwpenclosure2')[0].checked = false;
			}
		});
	});
	</script>";
}

function podlovewebplayer_enclosure_detect() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['enclosure_detect'])) {
		$checked = "checked ";
	}
	print "<input id='pwpenclosure1' name='podlovewebplayer_options[enclosure_detect]'
		$checked type='checkbox' value='1' />";
}

function podlovewebplayer_enclosure_force() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['enclosure_force'])) {
		$checked = "checked ";
	}
	print "<input id='pwpenclosure2' name='podlovewebplayer_options[enclosure_force]'
		$checked type='checkbox' value='1' />&nbsp;&nbsp;
		(additionally to manually shortcoded Podlove Web Players, if both are present)";
}

function podlovewebplayer_enclosure_richplayer() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['enclosure_richplayer'])) {
		$checked = "checked ";
	}
	print "<input id='pwpenclosure3' name='podlovewebplayer_options[enclosure_richplayer]'
		$checked type='checkbox' value='1' />&nbsp;&nbsp;
		(Use advanced player for enclosures)";
}


function podlovewebplayer_enclosure_bottom() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['enclosure_bottom'])) {
		$checked = "checked ";
	}
	print "<input id='pwpenclosure4' name='podlovewebplayer_options[enclosure_bottom]'
		$checked type='checkbox' value='1' />&nbsp;&nbsp;
		(instead of the top)";
}

function podlovewebplayer_chapter_height() {
	$options = get_option('podlovewebplayer_options');
	if(!isset( $options['chapter_height'])) {
		$options['chapter_height'] = "";
	}
	print "<input id='chapter_height' name='podlovewebplayer_options[chapter_height]'
		value='" . $options['chapter_height'] . "' style='width:3em;' /> px&nbsp;&nbsp;(keep empty to show all)";
}

function podlovewebplayer_style() {
	print "<div class='wrap'><h2>Custom CSS Style</h2>";
}

function podlovewebplayer_style_custom() {
	$options = get_option('podlovewebplayer_options');
	if (!isset($options['style_custom'])) {
		$options['style_custom'] = file_get_contents(plugin_dir_url(__FILE__) . 'static/default.css');
	}
	print "<textarea name='podlovewebplayer_options[style_custom]' id='pwpstyle1' dir='ltr' style='display:none;'>".$options['style_custom']."</textarea><script language='javascript'></script><p></p>
<div class='colorslider'><div id='color1' class='box'>
	<div id='colorboxes'></div>
	<div id='colorsliders'>
		<div><label for='hue'>Hue</label><input id='hue' onchange='pwpdcolorize();' name='hue' type='range' max='360' min='0'></div>
		<div><label for='sat'>Saturation</label><input id='sat' onchange='pwpdcolorize();' name='sat' type='range' max='100' min='0'></div>
		<div><label for='lum'>Luminance</label><input id='lum' onchange='pwpdcolorize();' name='lum' type='range' max='100' min='0'></div>
		<div><label for='gra'>Gradient</label><input id='gra' onchange='pwpdcolorize();' name='gra' type='range' max='20' min='0'></div>
	</div>
	<div><input type='button' onclick='pwpdexpertmode();' class='button' value='expert mode' /> <input type='button' onclick='pwpdinsertcolor();' class='button' value='enter color' /> <input type='button' onclick='pwpdrandomcolor();' class='button' value='random' /> <input type='button' onclick='pwpdcolorreset();' class='button' value='reset' /> <input name='Submit' type='submit' class='button button-primary' value='save'/></div><br/>
</div></div></div>";
	print '<audio id="pwpexample">
			<source src="http://podlove.github.com/podlove-web-player/samples/podlove-test-track.mp4" type="audio/mp4"></source>
			<source src="http://podlove.github.com/podlove-web-player/samples/podlove-test-track.mp3" type="audio/mpeg"></source>
			<source src="http://podlove.github.com/podlove-web-player/samples/podlove-test-track.ogg" type="audio/ogg; codecs=vorbis"></source>
			<source src="http://podlove.github.com/podlove-web-player/samples/podlove-test-track.opus" type="audio/ogg; codecs=opus"></source>
		</audio>
		<script>

		</script>';
	$podlovewebplayer_custompwpstyle = podlovewebplayer_custompwpstyle();
	if (!empty($podlovewebplayer_custompwpstyle)) {
		podlovewebplayer_makecss();
	} elseif (empty($podlovewebplayer_custompwpstyle) && file_exists(podlovewebplayer_css_path())) {
		unlink(podlovewebplayer_css_path());
	}
}

function podlovewebplayer_style_values() {
	$options = get_option('podlovewebplayer_options');
	if ( !isset( $options['style_values'] ) ) {
		$options['style_values'] = "{'hue':180,'sat':0,'lum':33,'gra':9}";
	}
	print "<input id='pwpconsole' name='podlovewebplayer_options[style_values]' value='".$options['style_values']."' style='width:19em;' />";
}

function podlovewebplayer_style_version() {
	$options = get_option('podlovewebplayer_options');
	if ( !isset( $options['style_version'] ) ) {
		$options['style_version'] = 1;
	} else {
		$options['style_version'] = $options['style_version']+1;
	}
	print $options['style_version'].'<input id="pwpcustomstyleversion" name="podlovewebplayer_options[style_version]" value="'.$options['style_version'].'" style="display:none;" />';
}


function podlovewebplayer_buttons() {
	print '<p>Here you can select, which buttons will be displayd. The Chapter-Toggle- and Summary-Info-Button are not configurable here, because they automaticle hidden, when no chapters/summary are provided.</p>'."\n\n";
}

function podlovewebplayer_buttons_time() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['buttons_time'])) {
		$checked = "checked ";
	}
	print "<input id='pwpbuttons1' name='podlovewebplayer_options[buttons_time]'
		$checked type='checkbox' value='1' />&nbsp;&nbsp;";
}

function podlovewebplayer_buttons_share() {
	$options = get_option('podlovewebplayer_options');
	$checked = "";
	if (isset($options['buttons_share'])) {
		$checked = "checked ";
	}
	print "<input id='pwpbuttons3' name='podlovewebplayer_options[buttons_share]'
		$checked type='checkbox' value='1' />&nbsp;&nbsp;";
}

function podlovewebplayer_info() {
	$scriptname = explode('/wp-admin', $_SERVER["SCRIPT_FILENAME"]);
	$dirname    = explode('/wp-content', dirname(__FILE__));
	$pluginfile = dirname(__FILE__) . '/podlove-web-player.php';
	$plugin_data = get_plugin_data( $pluginfile );

	print '<p>This is <strong>Version '.$plugin_data['Version'].'</strong> of the <strong>Podlove Web Player</strong>.<br>
	The <strong>Including file</strong> is: <code>wp-admin'.$scriptname[1].'</code><br>
	The <strong>PWP-directory</strong> is: <code>wp-content'.$dirname[1].'</code></p>
	<p>Want to contribute? Found a bug? Need some help? <br/>you can found our github repo/page at
	<a href="https://github.com/podlove/podlove-web-player">github.com/podlove/podlove-web-player</a></p>
	<p>You can\'t help with code? Ok, you can help with money, too. Go to
	<a href="http://podlove.org/donations/">podlove.org/donations/</a> to donate to the Podlove Project.</p>
	<p>We have <a href="https://github.com/podlove/podlove.github.com/wiki">lists with Podlove Publisher
	and Podlove Web Player using podcasts</a>, feel free to enter your podcast.</p>
	<p>If you found a bug, please tell us your WP- and PWP- (and PPP- if you use PPP) Version. <br/>Also your
	Browser version, your PHP version and the URL of your Podcast can help us, find the bug.</p>';
}

?>
